src/libfetchers/git.cc

```diff
@@ -23,6 +23,12 @@ static bool isNotDotGitDirectory(const Path & path)
     return not std::regex_match(path, gitDirRegex);
 }
 
+static bool isBareGitRepository(const Path & path)
+{
+    auto output = runProgram("git", true, { "-C", path, "rev-parse", "--is-bare-repository" });
```

Comment:
Calling an external program seems a bit expensive. We can also just do `return !pathExists(path + "/.git")` right?

Reply:
That seems reasonable, patch updated. Just note that this will have false positives on paths that are not git repos.